tiledb/pod/Cargo.toml

```diff
@@ -0,0 +1,33 @@
+[package]
+name = "tiledb-pod"
+edition.workspace = true
+rust-version.workspace = true
+version.workspace = true
+
+[dependencies]
+itertools = { workspace = true }
+num-traits = { workspace = true, optional = true }
+proptest = { workspace = true, optional = true }
+serde = { workspace = true, optional = true }
+serde_json = { workspace = true, optional = true }
+strategy-ext = { workspace = true, optional = true }
+thiserror = { workspace = true }
+tiledb-common = { workspace = true }
```

Comment:
Just a note to self that this hasn't removed the libtiledb.so linkage since we've got this transitive dependency. Though it looks pretty straightforward to cut that tie in a follow up PR that I assume is probably already half written waiting for me to approve this one. So on with the show!